tests/it/code_vsi.rs

```diff
@@ -128,3 +131,43 @@ int main() {
     );
     assert_eq!(Some(&expected), combined.get(Kind::CommentStrippedSha256));
 }
+
+#[test]
+fn strum_serde_fingerprint_kind_serialization_matches() {
+    for kind in Kind::iter() {
+        let json_value = serde_json::to_value(kind).expect("serde serialization");
+        let json = json_value.as_str().expect("get string");
+        let strum = kind.to_string();
+
+        assert_eq!(json, strum);
+    }
+}
+
+#[test]
+fn serde_serialization_matches_external_contract() {
+    let expected_serializations_for_kinds = vec![
+        "v1.class.jar",
+        "v1.mavencentral.jar",
+        "v1.raw.jar",
+        "comment_stripped:sha_256",
+        "sha_256",
+    ]
+    .into_iter()
+    .map(|s| s.to_string())
+    .collect::<HashSet<String>>();
+
+    let mut actual_serializations_for_kinds = HashSet::new();
+
+    for kind in Kind::iter() {
+        let json_value = serde_json::to_value(kind).expect("serde serialization");
+        let json = json_value.as_str().expect("get string").to_string();
+
+        // Check that the insert wasn't a duplicate.
+        assert!(actual_serializations_for_kinds.insert(json));
```

Comment:
if you do this, you get the effect of the comment in both the test failure (should it fail) and in code!
```suggestion
        assert!(
            actual_serializations_for_kinds.insert(json),
            "serialized value for kind {kind:?} is duplicate"
        );
```

Reply:
Ah thanks! I was looking for something like that!